### PR TITLE
Add Graduation Tips Document

### PR DESCRIPTION
**Pull request by Elzawawy**
- adds one file: `graduation-project-tips.md`. 
- the document contents includes 10 tips (each organized into an independent section).

### CODE REVIEW

File: graduation-project-tips.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# Ten Things they don't tell you about your graduation project
+
+In this document, I try to list the most important tips and tricks (in my opinion) in the hopes of helping future CCEians by sharing my experience with the graduation project and how to get the most out of it (hopefully).
+
+## Tip #1: Team Comes First :two_men_holding_hands:
+
+The team is the **most important factor** of any project, a team determines how the project starts, how the project ends and pretty much everything in between. 
+You must be comfortable to be yourself with your team and feel among friends, if this condition is not achieved, there will be fights and fights is not good for work.
```
Comment (Youssefares) [inline graduation-project-tips.md]: No need to be so negative I think. This is a bit extreme. Let's maybe advise people to be more professional and learn to work with people with a different mindset than theirs, but to expect less chemistry when working with people they haven't worked before.

File: graduation-project-tips.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+This is based on true events, it happened in my batch, in previous batches and will happen sadly in future batches (unless everyone reads/applies this tip hopefully).
```
Comment (Youssefares) [inline graduation-project-tips.md]: Also too gloomy :) Let's send a more positive message and focus on action points.

File: graduation-project-tips.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+- What techanical fields of study your project is tackling:question: Is it an embedded system project or a web/mobile based project:question:
+
+- What are the hot topics in that field of study we're tackling:question: What problems are related to that field in general:question:  
+
+- What technology stacks/fields your team collectively have of knowledge of:question:  What new techology stacks/fields are you required to learn:question:
+
+Try to answer those questions, they will narrow down your search space at least. You can lookup startup ideas that have previously explored that domain before and find something similar interesting to you.
+Also keep in mind, your idea need not to be a fresh idea. You can choose to re-create an already existing solution in a better fashion/ with new fancy features than those done in the field (Consider Careem company's success while Uber was already there).
+
+As a final note here, bring up more than one idea (one at least as backup plan) in case professors don't find this idea attractive enough for them, you should have other options in your bag.
```
Comment (Youssefares) [inline graduation-project-tips.md:33]: Might be worth mentioning some professors expect strong proposals and clear ideas from the get go and other are willing to brainstorm together.

File: graduation-project-tips.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+Again, knowledge of such tools will be a great qualification for work later on, since not all graudates learn about these stuff before they actually start working.
+
+## Tip #5: Learning Before Everything :books:
+
+This one is important because it's easy to forget actually I did myself and needed someone to remind me with it each time. The graduation project is your learning environment, it's analogous to a course that you put it's content/assignments and study it well. 
+What does this mean? 
+
+- **Don't** pick technology stacks because you're already familiar with them. If there's a stack (new to you) but is a better choice, take the chance and learn it. (I only knew 20% of the knowledge I gained after the year)
+- **Don't** rush tasks, this isn't a product that you're so eager to release then take care later of fine details, rather take your time and perfect each task you work on.
```
Comment (Youssefares) [inline graduation-project-tips.md]: This is not necessarily true, I think. It varies by context.

File: graduation-project-tips.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+You'll need to document everything that happens in the project life-time. It's a one-year project, don't expect to never forget stuff, on the contrary, you'll always forget details especially finer ones.
+What to document ?
+
+- Requirements Analysis, and project-planning details.
+- Supervisor notes in each meeting with him/her (Meeting Notes).
+- How did you reach an agreement on something (Brainstorming sessions).
+- Links/resources you find along the way (Bugs do re-appear).
+
+## Tip #7: Manage Code Smartly :octocat:
```
Comment (Youssefares) [inline graduation-project-tips.md:81]: Might be better to link to an external guide here for clean git usage and git tips. Just to make this guide a bit shorter.

File: graduation-project-tips.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+Everyone forgets the precious treasure lying in their hands until they lose it. If you have an academic email, you get free access to a lot of software tools that will be very beneficial in your graduation project. 
+The biggest gain comes from the awesome **[Github Student Pack](https://education.github.com/pack)**, just check the amount of tools you get access to or trials at least (From free tutorials/courses to AWS credits to perform learning/computations on machines).
+If any of the team members don't have such an email, I highly advise him.her to go get one asap because you'll want to use it.
+
+
+## Tip #9: Always Think Big ! :dart:
+
+Take this project seriously, it's going to (should) be the biggest project in your career so far, it will be the center of traction everyone will ask you about. If I am recruiting fresh graduates, that would be my first point to ask about technically.
+Don't just say this is another college project, because you may not get the chance to work on such collaborative project where you build everything from scratch again.
```
Comment (Youssefares) [inline graduation-project-tips.md]: No need to negative again and assume the reader is negligent :) Also may not necessarily be the biggest project in their career. Let's just focus on how big an opportunity it is without making a lot of strong statements.

**Review by Youssefares — CHANGES_REQUESTED** [review]
There is a lot of great advice on here. Thanks a lot for the initiative on this!

I left a few minor comments. One general comment though is that this is meant to be a guide not an opinion piece. So try to omit or phrase the bits where you say it's your personal opinion differently.

Another general comment we can optionally address is when talking about very common advice such as using git, documentation, planning tools. Linking to external articles and keeping the paragraphs short would help readers more.

File: graduation-project-tips.md (hunk 3 above)
Comment (Elzawawy) [reply graduation-project-tips.md:33]: Added at the end of Tip #3

File: graduation-project-tips.md (hunk 5 above)
Comment (Elzawawy) [reply graduation-project-tips.md:81]: Added am external guide. Here it is https://nvie.com/posts/a-successful-git-branching-model/

**Pull request merged**